### DISCUSSION TITLE
Only one message when the status of the counter changes

### DISCUSSION THREAD

**Discussion by rholze** [arendst/Tasmota#15300] [Q&A] [answered]
Hello,
I have a problem that I have not been able to solve on my own. Maybe there is already a solution here, but unfortunately I could not find it. 
I would like to use a water flow meter in my camper to turn on the water pump when the water tap is opened and the water flows. The water flow meter gives pulses (444/l) when the water flows. I have created a rule for this that works quite well. Unfortunately the rule is repeated with every impulse, but I only want a MQTT message when the event changes. So only when impulses come or no further impulses come. I just can't manage that. 

This is my current rule: 
```
Rule
ON COUNTER#C1>%var1% DO Backlog var1 %value%; publish tele/van1/WATERPUMP ON ENDON
ON COUNTER#C1=%var1% DO Backlog var1 %value%; publish tele/van1/WATERPUMP OFF ENDON
```
The result is:
![grafik](https://user-images.githubusercontent.com/100425033/161810398-db3c7078-58c3-49f5-9e27-53aac74a82d2.png)

I also tried something like this, to simulate a hysteresis:
```
Rule2
ON COUNTER#C1>%var1% DO Backlog var1 %value%; publish tele/van1/WATERPUMP ON; var2 %value%; add1 2; sub2 2 ENDON
ON %var2%<COUNTER#C1 DO Backlog var2 %value%; publish tele/van1/WATERPUMP OFF; var1 %value%; add1 2; sub2 2 ENDON
```
The result is depressing:
![grafik](https://user-images.githubusercontent.com/100425033/161816326-b121a240-c267-4a6d-a045-57c2b86a72c2.png)

As you can see, I haven't been working with Tasmota commands/rules for that long. So any help is welcome.
Thanks a lot Ronald

**Comment by sfromis** [comment, marked as answer]
Using the `Rule` command, you set a rule set as one-shot, meaning that a trigger does not fire again while the condition is still met.

`%var2%<COUNTER#C1` - IIRC, the trigger name must be to the left, this is nothing like a programming language where you can just move things around.
